ghostty: Update to 1.2.0

infinitymdm opened the pull request:
**Summary**

Update ghostty to the latest release. 

This PR also includes updates to zig (fixes #5257) and the initial inclusion of gtk4-layer-shell, a dependency of ghostty.

**Test Plan**

Launch ghostty. Make sure it works as expected with starship, fish, etc. Test custom keybinds.

**Checklist**

- [x] Package was built and tested against unstable
- [ ] This change could gainfully be listed in the weekly sync notes once merged  <!-- Write an appropriate message in the Summary section, then add the "Topic: Sync Notes" label -->

infinitymdm commented:
> Does `ncdu` also need an update to work with the new Zig?

Well, bumping & rebuilding with the current package.yml doesn't seem to work, so yeah, probably. Is there a tool to search for other packages which require zig as a builddep and might potentially need rebuilds?

EbonJaeger commented:
I just ripgrep the entire package tree.

```bash
rg "zig" packages/*/*/package.yml
```

EbonJaeger commented:
Upgrading `ncdu` to 2.9.1 Just Works :tm:

```yaml
name       : ncdu
version    : 2.9.1
release    : 25
source     :
    - https://dev.yorhel.nl/download/ncdu-2.9.1.tar.gz : bfd1094e1400ee89cfd59200eab940f025ccdc0c238067105c984a3c4857bf7e
homepage   : https://dev.yorhel.nl/ncdu
license    : MIT
component  : system.utils
summary    : Disk usage analyzer with ncurses interface
description: |
    ncdu is a disk usage analyzer with an ncurses interface, aimed to be run on a remote server over SSH. It is therefore designed to be fast, simple, and easy to use.
environment: |
    export ZIG_FLAGS="-Dcpu=baseline --release=fast"
builddeps  :
    - pkgconfig(libzstd)
    - pkgconfig(ncursesw)
    - zig
build      : |
    %make
install    : |
    install -D -m 00755 zig-out/bin/ncdu $installdir/usr/bin/ncdu
    install -D -m 00644 ncdu.1 $installdir/usr/share/man/man1/ncdu.1
```

infinitymdm commented:
Oh, I didn't realize that ncdu was the only other zig package. I was getting worried that I'd have to do a whole bunch of programs.

Added ncdu (thanks for the yml!) and updated. Disclaimer: I haven't done any extensive testing of ncdu outside of checking that it opens an ncurses tui.

infinitymdm commented on packages/g/gtk4-layer-shell/package.yml:
```diff
@@ -0,0 +1,24 @@
+builddeps  :
+    - pkgconfig(gobject-introspection-1.0)
+    - pkgconfig(gtk4)
+    - pkgconfig(luajit)
```
I think I took that from the AUR package, but you're right that it doesn't seem to be needed. Rebuilding ghostty now just to make sure.
